karabo_data/read_machinery.py

```diff
@@ -205,3 +205,80 @@ def find_proposal(propno):
         return d
 
     raise Exception("Couldn't find proposal dir for {!r}".format(propno))
+
+
+class VirtualStack:
+    """'Virtual' array for stacking detector data with modules on axis -3
+
+    Access is limited to either a single module at a time or all modules
+    together, but this is enough to assemble detector images.
+    """
+    def __init__(self, data, nmodules, mod_shape, dtype, fillvalue):
+        self._nmodules = nmodules
```

Comment:
Should this have a default value which is None (`data.shape[0]`)?

Reply:
As it's only constructed somewhere where we already have that number, I prefer to always pass it explicitly. If any of the modules are missing data, you want `nmodules > len(data)`.

Reply:
OK! I see. Thanks!